### PR TITLE
Use detected default compiler

### PR DESCRIPTION
**Pull request by Colengms**
Addresses: https://github.com/microsoft/vscode-cpptools/issues/5776

The detected default compiler and associated values were used to populate newly created configurations, but were not used as a fallback if the compiler was unspecified.

This change handles the situation where no compiler was specified in the c_cpp_properties, or explicitly as a default setting.  Instead of using no/blank compiler, the detected compiler and associated values are used.

### CODE REVIEW

File: Extension/src/LanguageServer/configurations.ts [hunk 1]
```diff
@@ -628,6 +628,27 @@ export class CppProperties {
             configuration.cStandard = this.updateConfigurationString(configuration.cStandard, settings.defaultCStandard, env);
             configuration.cppStandard = this.updateConfigurationString(configuration.cppStandard, settings.defaultCppStandard, env);
             configuration.intelliSenseMode = this.updateConfigurationString(configuration.intelliSenseMode, settings.defaultIntelliSenseMode, env);
+            if ((configuration.compilerPath === undefined || configuration.compilerPath === "") && !!this.defaultCompilerPath) {
```
Comment (sean-mcmanus) [inline Extension/src/LanguageServer/configurations.ts]: It doesn't seem like "" should use the default, right?

File: Extension/src/LanguageServer/configurations.ts [hunk 2]
```diff
@@ -628,6 +628,27 @@ export class CppProperties {
...
+                configuration.compilerPath = this.defaultCompilerPath;
+                if ((!configuration.cStandard || configuration.cStandard === "") && !!this.defaultCStandard) {
```
Comment (sean-mcmanus) [inline Extension/src/LanguageServer/configurations.ts]: Is it intentional to condition these on the compilerPath?

Comment (Colengms) [reply Extension/src/LanguageServer/configurations.ts]: Yes.  If we're not actually using the detected default compiler, we should not apply the detected default compiler's standards, intelliSenseMode, etc.  Right?

Comment (Colengms) [reply Extension/src/LanguageServer/configurations.ts]: But, I'm not sure about defaultWindowsSdkVersion, defaultIncludes, defaultFrameworks .

File: Extension/src/LanguageServer/configurations.ts [hunk 3]
```diff
@@ -628,6 +628,27 @@ export class CppProperties {
             configuration.cStandard = this.updateConfigurationString(configuration.cStandard, settings.defaultCStandard, env);
             configuration.cppStandard = this.updateConfigurationString(configuration.cppStandard, settings.defaultCppStandard, env);
             configuration.intelliSenseMode = this.updateConfigurationString(configuration.intelliSenseMode, settings.defaultIntelliSenseMode, env);
+            if (!configuration.compilerPath && !!this.defaultCompilerPath) {
+                configuration.compilerPath = this.defaultCompilerPath;
```
Comment (bobbrow) [inline Extension/src/LanguageServer/configurations.ts]: `!configuration.compilerPath` will be true when compilerPath is set to "".  We should not be overwriting `""` with `this.defaultCompilerPath` in that case. That will break folks who depend on the `""` behavior.

**Review by bobbrow — APPROVED** [review]

**Pull request merged**